File: android-SpaceSub/app/src/main/java/dev/squad52/spacesub/models/Subscription.kt
package dev.squad52.spacesub.models

import com.google.gson.annotations.SerializedName

enum class BillingCycle {
    @SerializedName("WEEKLY") WEEKLY,
    @SerializedName("MONTHLY") MONTHLY,
    @SerializedName("QUARTERLY") QUARTERLY,
    @SerializedName("YEARLY") YEARLY;

    val russianName: String
        get() = when (this) {
            WEEKLY -> "Еженедельно"
            MONTHLY -> "Ежемесячно"
            QUARTERLY -> "Ежеквартально"
            YEARLY -> "Ежегодно"
        }

    val shortLabel: String
        get() = when (this) {
            WEEKLY -> "нед"
            MONTHLY -> "мес"
            QUARTERLY -> "кв"
            YEARLY -> "год"
        }
}

data class Subscription(
    val id: String,
    val userId: String,
    val name: String,
    val description: String? = null,
    val amount: Double,
    val currency: String,
    val periodType: BillingCycle,
    val nextBilling: String? = null,
    val category: String? = null,
    val isActive: Boolean,
    val logoUrl: String? = null,
    val createdAt: String,
    val updatedAt: String
)

data class DetectedSubscription(
    val id: String,
    val merchant: String,
    val amount: Double,
    val currency: String,
    val periodType: BillingCycle,
    val lastChargeDate: String? = null,
    val nextExpectedCharge: String? = null,
    val isActive: Boolean,
    val confidence: Double,
    val transactionCount: Int
)

data class SubscriptionSummary(
    val activeCount: Int = 0,
    val monthlyTotal: Double = 0.0,
    val yearlyTotal: Double = 0.0,
    val upcomingNext7Days: List<DetectedSubscription> = emptyList()
)

data class CancelSubscriptionResponse(
    val cancelled: Boolean,
    val bankPaymentId: String?
)

File: ios-SpaceSub/ios-SpaceSub/Models/Subscription.swift
import Foundation

nonisolated enum BillingCycle: String, Codable, Sendable, CaseIterable {
    case weekly = "WEEKLY"
    case monthly = "MONTHLY"
    case quarterly = "QUARTERLY"
    case yearly = "YEARLY"

    var displayName: String {
        switch self {
        case .weekly: "Weekly"
        case .monthly: "Monthly"
        case .quarterly: "Quarterly"
        case .yearly: "Yearly"
        }
    }

    var russianName: String {
        switch self {
        case .weekly: "Еженедельно"
        case .monthly: "Ежемесячно"
        case .quarterly: "Ежеквартально"
        case .yearly: "Ежегодно"
        }
    }

    var shortLabel: String {
        switch self {
        case .weekly: "/ нед"
        case .monthly: "/ мес"
        case .quarterly: "/ кв"
        case .yearly: "/ год"
        }
    }
}

nonisolated struct Subscription: Codable, Identifiable, Sendable {
    let id: String
    let userId: String
    let name: String
    let description: String?
    let amount: Double
    let currency: String
    let periodType: BillingCycle
    let nextBilling: String
    let category: String?
    let isActive: Bool
    let logoUrl: String?
    let createdAt: String
    let updatedAt: String
}

nonisolated struct CreateSubscriptionRequest: Codable, Sendable {
    let name: String
    var description: String?
    let amount: Double
    var currency: String = "RUB"
    var periodType: BillingCycle = .monthly
    let nextBilling: String
    var category: String?
    var isActive: Bool = true
    var logoUrl: String?

    enum CodingKeys: String, CodingKey {
        case name, description, amount, currency
        case periodType = "billingCycle"
        case nextBilling, category, isActive, logoUrl
    }
}

nonisolated struct DetectedSubscription: Codable, Identifiable, Sendable {
    let id: String
    let merchant: String
    let amount: Double
    let currency: String
    let periodType: BillingCycle
    let lastChargeDate: String
    let nextExpectedCharge: String
    let isActive: Bool
    let confidence: Double
    let transactionCount: Int
    let logoUrl: String?
}

nonisolated struct CancelSubscriptionResponse: Codable, Sendable {
    let cancelled: Bool
    let bankPaymentId: String?
}

nonisolated struct ConfirmSuggestionResponse: Codable, Sendable {
    let subscription: Subscription
    let linkedTransactions: Int
}

nonisolated struct SubscriptionSummary: Codable, Sendable {
    let activeCount: Int
    let monthlyTotal: Double
    let yearlyTotal: Double
    let upcomingNext7Days: [DetectedSubscription]
}
